doc/cephfs/kernel.rst

```diff
@@ -1,58 +1,95 @@
-====================================
- Mount CephFS with the Kernel Driver
-====================================
+=================================
+ Mount CephFS using Kernel Driver
+=================================
 
-It is recommended to install the ``/sbin/mount.ceph`` kernel mount
-helper if working with the kernel cephfs driver. To mount CephFS with
-the kernel driver you may use the ``mount`` command::
+Prerequisite
+------------
+Before mounting CephFS, ensure that the client host (where CephFS will be
+mounted and used) has a copy of the keyring for the user which has CAPS for
+the MDS. Keyring must be present on the host where the Ceph Monitor resides.
 
-	# mkdir /mnt/mycephfs
-	# mount -t ceph :/ /mnt/mycephfs
+#. Copy the Ceph keyring from the MON host to client host at ``/etc/ceph``::
 
-Omitting the monitor addresses will cue ``mount.ceph`` to look them up
-in the local configuration file.  If you know at least one of the monitor
-addresses and ports you can specify them directly in a comma-separated
-list, and the kernel will avoid looking them up::
+    # on client host
+    mkdir /etc/ceph
+    scp {user}@{server-machine}:/etc/ceph/ceph.keyring /etc/ceph/ceph.keyring
 
-        # mount -t ceph 192.168.0.1:6789,192.168.0.2:6789:/ /mnt/mycephfs
+#. Ensure that the keyring has appropriate permissions::
 
-To mount a subtree of the cephfs root, append the path to the device
-string ::
+    chmod 600 /etc/ceph/ceph.keyring
 
-        # mount -t ceph :/subvolume/dir1/dir2  /mnt/mycephfs -o name=fs
+# It is recommeneded to have mount.ceph helper installed. Verify that it is
+  installed::
 
-To mount the Ceph file system with ``cephx`` authentication enabled, the
-kernel must authenticate with the cluster. The default ``name=`` option
-is ``guest``.  The mount.ceph helper will automatically attempt to find
-a secret key in a cephx keyring if it's not specified. For example, to
-mount the filesystem as the cephx user ``fs``::
+    ls /sbin/mount.ceph
 
-        # mount -t ceph :/ /mnt/mycephfs -o name=fs
+# You may also get the Ceph configuration file on the client host, although
+  this is optional::
```

Comment:
It's optional, but if you don't have it then you have to provide mon addrs yourself. What we might want to do here is have this document the way we recommend (with a local config file and keyring), and add a section at the end that explains how to do this providing the mon addrs and secrets manually.

Reply:
> It's optional, but if you don't have it then you have to provide mon addrs yourself. 

Yeah, the prerequisites are actual are optional. I've mentioned that too.

> What we might want to do here is have this document the way we recommend (with a local config file and keyring), and add a section at the end that explains how to do this providing the mon addrs and secrets manually.

I wanted to keep the recommended command and the actual complete closer so that the reader can better relate how actually mount command works.